core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -459,6 +459,11 @@ public void setResponseTimeoutMillis(long responseTimeoutMillis) {
             clearResponseTimeout();
         }
 
+        if (this.responseTimeoutMillis == 0 && responseTimeoutMillis > 0) {
```

Comment:
`responseTimeoutMillis > 0` is always true at this point.

Reply:
Oops..